src/algorithms/optimization/peps_optimization.jl

```diff
@@ -144,6 +201,136 @@ function fixedpoint(
     return peps_final, env_final, cost, info
 end
 
+"""
+    function select_algorithm(
+        ::typeof(fixedpoint),
+        env₀::CTMRGEnv;
+        tol=Defaults.optimizer_tol,
+        verbosity=2,
+        boundary_alg=(;),
+        gradient_alg=(;),
+        optimization_alg=(;),
+        (finalize!)=OptimKit._finalize!,
+    )
+
+Parse optimization keyword arguments on to the corresponding algorithm structs and return
+a final `PEPSOptimize` to be used in `fixedpoint`. For a description of the keyword
+arguments, see [`fixedpoint`](@ref).
+"""
+function select_algorithm(
+    ::typeof(fixedpoint),
+    env₀::CTMRGEnv;
+    tol=Defaults.optimizer_tol, # top-level tolerance
+    verbosity=2, # top-level verbosity
+    boundary_alg=(;),
+    gradient_alg=(;),
+    optimization_alg=(;),
+    (finalize!)=OptimKit._finalize!,
+)
+
+    # top-level verbosity
+    if verbosity ≤ 0 # disable output
+        boundary_verbosity = -1
+        gradient_verbosity = -1
+        optimizer_verbosity = -1
+    elseif verbosity == 1 # output only optimization steps and degeneracy warnings
+        boundary_verbosity = -1
+        gradient_verbosity = -1
+        optimizer_verbosity = 3
+    elseif verbosity == 2 # output optimization and boundary information
+        boundary_verbosity = 2
+        gradient_verbosity = -1
```

Comment:
It would be nice to have some gradient verbosity without having the full boundary verbosity, just to get the warnings if the gradient fails to converge. So I think it would be good to either add another level which does this, or put the gradient verbosity to `1` here.